test/externalTests/anvil.js

```diff
@@ -0,0 +1,120 @@
+    const book = anvil.findInventoryItem(mcData.itemsByName.enchanted_book.id)
+
+    await anvil.combine(sword, book)
+    await bot.test.wait(1000)
+    // test result
+    assert.strictEqual(bot.experience.level, 994)
+    assert.strictEqual(bot.inventory.slots[9].repairCost, 1)
+    assert.deepStrictEqual(bot.inventory.slots[9].enchants, [{ name: 'sharpness', lvl: 5 }])
+  }
+  async function testTwo () { // combining two items in inventory, but there are three items, so this is more a test of using nbt when picking the item in inventory
+    bot.chat(`/clear ${bot.username}`)
+    await bot.test.becomeCreative()
+
+    await bot.test.setInventorySlot(36, new Item(mcData.itemsByName.diamond_sword.id, 1))
+    await bot.test.setInventorySlot(37, makeItem({ type: mcData.itemsByName.diamond_sword.id, enchants: [{ name: 'sharpness', lvl: 5 }] }))
+    await bot.test.setInventorySlot(38, makeBook([{ name: 'unbreaking', lvl: 3 }]))
+
+    await bot.test.becomeSurvival()
+
+    const anvil = await bot.openAnvil(b)
+
+    const sword = bot.inventory.slots[37]
+    const book = anvil.findInventoryItem(mcData.itemsByName.enchanted_book.id)
+
+    await anvil.combine(sword, book)
+    await bot.test.wait(1000)
+    // test result
+    assert.strictEqual(bot.experience.level, 991)
+    assert.strictEqual(bot.inventory.slots[9].repairCost, 1)
+    assert.deepStrictEqual(bot.inventory.slots[9].enchants, [{ name: 'sharpness', lvl: 5 }, { name: 'unbreaking', lvl: 3 }])
+  }
+
+  async function testThree () { // using anvil.rename
+    bot.chat(`/clear ${bot.username}`)
+    await bot.test.becomeCreative()
+
+    await bot.test.setInventorySlot(36, makeItem({ type: mcData.itemsByName.diamond_sword.id }))
+
+    await bot.test.becomeSurvival()
+
+    const anvil = await bot.openAnvil(b)
+
+    const sword = anvil.findInventoryItem(mcData.itemsByName.diamond_sword.id)
+    await anvil.rename(sword, 'hello')
+    await bot.test.wait(1000)
+    // test result
+    assert.strictEqual(bot.experience.level, 990)
+    assert.strictEqual(bot.inventory.slots[9].repairCost, renameCost())
+    assert.deepStrictEqual(bot.inventory.slots[9].customName, renameName('hello'))
+  }
+
+  async function testFour () { // test 2 + a rename
+    bot.chat(`/clear ${bot.username}`)
+    await bot.test.becomeCreative()
+
+    await bot.test.setInventorySlot(36, new Item(mcData.itemsByName.diamond_sword.id, 1))
+    await bot.test.setInventorySlot(37, makeItem({ type: mcData.itemsByName.diamond_sword.id, enchants: [{ name: 'sharpness', lvl: 5 }] }))
+    await bot.test.setInventorySlot(38, makeBook([{ name: 'unbreaking', lvl: 3 }]))
+
+    await bot.test.becomeSurvival()
+
+    const anvil = await bot.openAnvil(b)
+
+    const sword = bot.inventory.slots[37]
+    const book = anvil.findInventoryItem(mcData.itemsByName.enchanted_book.id)
+
+    await anvil.combine(sword, book, 'lol')
+    await bot.test.wait(1000)
```

Comment:
should be removed (replaced by an await if needed)

Reply:
fixed